File: Sample/BasicSample/sample_cs/GeometryObject2D/GeometryObject2D_ArcShape.cs
/// <summary>
/// 弧を表示するサンプル。
/// </summary>
class GeometryObject2D_ArcShape : ISample
{
    public string Description
    {
        get { return ""; }
    }
	public string Title
	{
		get { return ""; }
	}
	public string ClassName
	{
		get { return "GeometryObject2D_ArcShape"; }
	}

    public void Run()
    {
        // Altseedを初期化する。
        asd.Engine.Initialize("GeometryObject2D_ArcShape", 640, 480, new asd.EngineOption());

        // 図形描画クラスのコンストラクタを呼び出す。
        var geometryObj = new asd.GeometryObject2D();

        // 図形描画オブジェクトのインスタンスをエンジンに追加する。
        asd.Engine.AddObject2D(geometryObj);

        // 弧の図形クラスのインスタンスを生成する。
        var arc = new asd.ArcShape();

        // 弧の外径、内径、頂点数、中心位置、開始頂点番号、終了頂点番号を指定する。
        arc.OuterDiameter = 400;
        arc.InnerDiameter = 40;
        arc.NumberOfCorners = 96;
        arc.Position = new asd.Vector2DF(320, 240);
        arc.StartingCorner = 90;
        arc.EndingCorner = 45;

        // 弧を描画する図形として設定する。
        geometryObj.Shape = arc;

        // Altseedを更新する。
        while (asd.Engine.DoEvents())
        {
            // Altseedを更新する。
            asd.Engine.Update();

            Recorder.TakeScreenShot("GeometryObject2D_ArcShape", 30);
        }

        // Altseedの終了処理をする。
        asd.Engine.Terminate();
    }
}

File: Sample/BasicSample/sample_cpp/GeometryObject2D/GeometryObject2D_ArcShape.cpp
#include <Base.h>

/**
@brief	弧を表示するサンプル。
*/
void GeometryObject2D_ArcShape()
{
	// Altseedを初期化する
	asd::Engine::Initialize(asd::ToAString("GeometryObject2D_ArcShape").c_str(), 640, 480, asd::EngineOption());

	// 図形描画オブジェクトのインスタンスを生成する。
	auto geometryObj = std::make_shared<asd::GeometryObject2D>();

	// 図形描画オブジェクトのインスタンスをエンジンに追加する。
	asd::Engine::AddObject2D(geometryObj);

	// 弧を図形描画クラスにて描画する。
	auto arc = std::make_shared<asd::ArcShape>();

	// 弧の外径、内径、頂点数、中心位置、開始頂点番号、終了頂点番号を指定。
	arc->SetOuterDiameter(400);
	arc->SetInnerDiameter(40);
	arc->SetNumberOfCorners(96);
	arc->SetPosition(asd::Vector2DF(320, 240));
	arc->SetStartingCorner(90);
	arc->SetEndingCorner(45);

	// 弧を描画する図形として設定する。
	geometryObj->SetShape(arc);


	// Altseedのウインドウが閉じられていないか確認する。
	while (asd::Engine::DoEvents())
	{
		// Altseedを更新する。
		asd::Engine::Update();
	}

	// Altseedを終了する。
	asd::Engine::Terminate();
}